src/cloud/lib/editor/CodeMirror.ts

```diff
@@ -21,3 +22,239 @@ export interface EditorPosition {
   line: number
   ch: number
 }
+
+// commented out are available but in later versions of CodeMirror - enable when available
```

Comment:
Which version of codemirror do we need?

Reply:
Not sure will check today

Reply:
I checked, I could not find when they added those modes, maybe they are available with different names, but I also found some race condition bug which was fixed in 5.61.0 (see releases https://codemirror.net/doc/releases.html) for show-hint addon.

Reply:
I test updated to latest v5 version, didn't see any additional modes, maybe those are for v6 code mirror.